orttraining/tools/scripts/opset12_model_transform.py

```diff
@@ -0,0 +1,133 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+#
+# This converter is an internal util to upgrade existing bert/gpt-2 models, 
+# which were previously transformed/optimized from orginal model, to Opset 12 
+# version as well as replacing deprecated node, i.e., TrainableDropout with 
+# the "Dropout" node matching the Opset 12 Spec. Typically, a model to 
+# be run by this scripts would have "_optimized" substring in its model name, 
+# and the graph should have one or more "TrainableDropout" nodes in its graph. 
+# Example usage: 
+#   python opset12_model_transform.py bert-base-uncased_L_12_H_768_A_12_V_30528_S_512_Dp_0.1_optimized_layer_norm.onnx
+# Output: 
+#   bert-base-uncased_L_12_H_768_A_12_V_30528_S_512_Dp_0.1_optimized_layer_norm_opset12.onnx
+
+import sys
+import onnx
+from onnx import helper, shape_inference
+from onnx import TensorProto
+import numpy as np
+from onnx import numpy_helper
+
+if len(sys.argv) < 2:
+    print("Please give model path...")
+    exit(1)
+
+input_model_name = sys.argv[1]
+output_model_name = input_model_name[:-5] + '_opset12.onnx'
+
+model = onnx.load(input_model_name)
+
+def find_input_node(model, arg):
```

Comment:
>find_input_node [](start = 4, length = 15)

this function is called find_input_node, but it looks through node outputs. i'm a bit confused by the name.

Reply:
Sorry about that, the code was inherited from previous one, I can include the change in my next PR to clarify more.